src/utils/graph.ts

```diff
@@ -620,3 +374,102 @@ const isPluralQuery = (key: EntityKey): boolean => {
 export const getSubgraphIdForProject = (pv: PV, projectId: number) => {
   return `${pv}-${projectId}`
 }
+
+export const parseBigNumberKeyVals = <T, K extends keyof T>(
+  json: { [k in K]: BigNumberish | null },
+  keys: K[],
+) => {
+  return keys.reduce(
+    (acc, k) => ({
+      ...acc,
+      ...(Object.hasOwn(json, k) ? parseBigNumberKeyVal(k, json[k]) : {}),
+    }),
+    {} as { [k in K]: BigNumber },
+  )
+}
+
+/**
+ * Parse a key value pair from an object
+ * @param key Name of key
+ * @param val Value to convert to BigNumber
+ * @returns Key value pair, where value is a BigNumber
+ */
+export const parseBigNumberKeyVal = <K extends string | number | symbol>(
```

Comment:
some of these utils aren't really graph specific. but, not sure there's a better place